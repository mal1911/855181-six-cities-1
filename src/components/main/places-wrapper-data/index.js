import PlacesWrapperData from "./places-wrapper-data";

export default PlacesWrapperData;
