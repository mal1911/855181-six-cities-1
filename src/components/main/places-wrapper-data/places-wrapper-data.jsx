import React from "react";
import Cards from "../cards";
import CardsHeader from "../cards-header";
import PopupSortingElement from "../pupup-sorting-element/popup-sorting-element";
import withPopupToggle from "../../../hocs/with-popup-toggle/with-popup-toggle";
import Map from "../../map";
import {connect} from "react-redux";
import {getOffersLocationsData} from "../../../reducer/data/selectors";
import {locationType} from "../../../prop-types";
import PropTypes from "prop-types";

const PlacesWrapperData = ({offersLocationsData}) => {
  const TogglePopupSortingElement = withPopupToggle(PopupSortingElement);
  return <React.Fragment>
    <div className="cities__places-wrapper">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <CardsHeader/>
          <TogglePopupSortingElement/>
          <Cards/>
        </section>
        <div className="cities__right-section">
          <Map
            className={`cities__map`}
            offersLocationsData={offersLocationsData}
          />
        </div>
      </div>
    </div>
  </React.Fragment>;
};

PlacesWrapperData.propTypes = {
  offersLocationsData: PropTypes.arrayOf(PropTypes.shape({
    location: locationType.isRequired,
    isActive: PropTypes.bool
  })),
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  offersLocationsData: getOffersLocationsData(state),
});

export {PlacesWrapperData};

export default connect(mapStateToProps)(PlacesWrapperData);
