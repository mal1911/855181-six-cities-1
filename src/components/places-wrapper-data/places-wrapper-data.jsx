import React from "react";
import Cards from "../cards/cards";
import CardsHeader from "../cards-header/cards-header";
import PopupSortingElement from "../pupup-sorting-element/popup-sorting-element";
import withActiveItem from "../../hocs/with-active-item/with-active-item";
import withPopupToggle from "../../hocs/with-popup-toggle/with-popup-toggle";
import Map from "../map/map";

const PlacesWrapperData = () => {
  const ActivatedPopupSortingElement = withActiveItem(withPopupToggle(PopupSortingElement), 0);
  const ActivatedCards = withActiveItem(Cards);

  return <React.Fragment>
    <div className="cities__places-wrapper">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <CardsHeader/>
          <ActivatedPopupSortingElement/>
          <ActivatedCards/>
        </section>
        <div className="cities__right-section">
          <Map/>
        </div>
      </div>
    </div>
  </React.Fragment>;
};

export default PlacesWrapperData;
