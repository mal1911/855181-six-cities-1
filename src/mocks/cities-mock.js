export const citiesMock = [
  {
    id: 1,
    name: `Paris`,
  },
  {
    id: 2,
    name: `Cologne`,
  },
  {
    id: 3,
    name: `Brussels`,
  },
  {
    id: 4,
    name: `Amsterdam`,
  },
  {
    id: 5,
    name: `Hamburg`,
  },
  {
    id: 6,
    name: `Dusseldorf`,
  },
];

