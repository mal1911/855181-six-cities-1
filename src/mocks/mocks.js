export const offersData = [
  {
    id: 1,
    isPremium: true,
    city: `Paris`,
    title: `Paris Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    isPremium: false,
    city: `Paris`,
    title: `Paris Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    isPremium: false,
    city: `Paris`,
    title: `Paris Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 4,
    isPremium: false,
    city: `Paris`,
    title: `Paris Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 5,
    isPremium: true,
    city: `Cologne`,
    title: `Cologne Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 6,
    isPremium: false,
    city: `Cologne`,
    title: `Cologne Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 7,
    isPremium: true,
    city: `Brussels`,
    title: `Brussels Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 8,
    isPremium: false,
    city: `Brussels`,
    title: `Brussels Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 9,
    isPremium: true,
    city: `Amsterdam`,
    title: `Amsterdam Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 10,
    isPremium: false,
    city: `Amsterdam`,
    title: `Amsterdam Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 11,
    isPremium: true,
    city: `Hamburg`,
    title: `Hamburg Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 12,
    isPremium: false,
    city: `Hamburg`,
    title: `Hamburg Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 13,
    isPremium: true,
    city: `Dusseldorf`,
    title: `Dusseldorf Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 14,
    isPremium: false,
    city: `Dusseldorf`,
    title: `Dusseldorf Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
];

export const citiesData = [
  {
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Cologne`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Brussels`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9],
  },
];
