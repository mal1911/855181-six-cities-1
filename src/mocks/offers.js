export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`
  },
  {
    title: `Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`
  },
  {
    title: `Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`
  },
];
