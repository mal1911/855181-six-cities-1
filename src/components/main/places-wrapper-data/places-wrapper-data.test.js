import React from "react";
import renderer from "react-test-renderer";
import {PlacesWrapperData} from "./places-wrapper-data";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {offersLocationsData, initialState} from "../../../mocks/mocks";

describe(`PlacesWrapperData correctly renders`, () => {
  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(initialState);
    tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <PlacesWrapperData
              offersLocationsData={offersLocationsData}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
  });

  it(`PlacesWrapperData correctly renders`, () => {
    expect(tree).toMatchSnapshot();
  });
});
