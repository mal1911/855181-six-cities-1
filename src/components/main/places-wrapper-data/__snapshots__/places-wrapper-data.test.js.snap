// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesWrapperData correctly renders PlacesWrapperData correctly renders 1`] = `
<div
  className="cities__places-wrapper"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        0
         places to stay in 
        
      </b>
      <form
        action="#"
        className="places__sorting"
        method="get"
        onKeyDown={[Function]}
      >
        <span
          className="places__sorting-caption"
        >
          Sort by Â 
        </span>
        <span
          className="places__sorting-type"
          onClick={[Function]}
          tabIndex="0"
        >
          Popular
          <svg
            className="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlinkHref="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          className="places__options places__options--custom null"
        >
          <li
            className="places__option  places__option--active"
            onClick={[Function]}
            tabIndex="0"
          >
            Popular
          </li>
          <li
            className="places__option null"
            onClick={[Function]}
            tabIndex="0"
          >
            Price: low to high
          </li>
          <li
            className="places__option null"
            onClick={[Function]}
            tabIndex="0"
          >
            Price: high to low
          </li>
          <li
            className="places__option null"
            onClick={[Function]}
            tabIndex="0"
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        className="cities__places-list places__list tabs__content"
      />
    </section>
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
        id="map"
      />
    </div>
  </div>
</div>
`;
